Python/import.c

```diff
@@ -221,16 +221,32 @@ import_ensure_initialized(PyInterpreterState *interp, PyObject *mod, PyObject *n
         rc = _PyModuleSpec_IsInitializing(spec);
         Py_DECREF(spec);
     }
-    if (rc <= 0) {
+    if (rc == 0) {
+        goto done;
+    }
+    else if (rc < 0) {
         return rc;
     }
+
     /* Wait until module is done importing. */
     PyObject *value = PyObject_CallMethodOneArg(
         IMPORTLIB(interp), &_Py_ID(_lock_unlock_module), name);
     if (value == NULL) {
         return -1;
     }
     Py_DECREF(value);
+
+done:
+    /* When -Ximporttime=2, print an import time entry even if an
```

Comment:
Is it right place to add an output? Where the line for non-cached import is printed?

Reply:
The original `-Ximporttime` output is in `import_find_and_load`. This PR adds output to `import_ensure_initialized`. 

The two branches are called in `PyImport_ImportModuleLevelObject`:

```
    if (mod != NULL && mod != Py_None) {
        if (import_ensure_initialized(tstate->interp, mod, abs_name) < 0) {
            goto error;
        }
    }
    else {
        Py_XDECREF(mod);
        mod = import_find_and_load(tstate, abs_name);
        if (mod == NULL) {
            goto error;
        }
    }
```